doc/man3/SSL_CTX_set1_curves.pod

```diff
@@ -33,6 +33,9 @@ SSL_get1_curves, SSL_get_shared_curve
  int SSL_get1_curves(SSL *ssl, int *curves);
  int SSL_get_shared_curve(SSL *s, int n);
 
+ int SSL_CTX_get0_implemented_groups(SSL_CTX *ctx, long all,
```

Comment:
Why is "all" a long - typically we have a int for simple boolean parameters - using long stands out as "odd"

Reply:
The SSL control API takes a long and a pointer, I could I guess document it taking an int, and cast to long before calling the control, function, but then being a macro it would accept **anything** that can be cast to a long. Though i could write:

```
#define warn_non_int(i) (0 ? snprintf(NULL, 0, "%d", (i)) : (i))
```
and use `(long) warn_non_int(arg)`.
Which would warn for any wrong types or potentially lossy conversions:

```c
...: warning: format ‘%d’ expects argument of type ‘int’, but argument 4 has type ‘long int’ [-Wformat=]
    6 |     int j = (0 ? snprintf(NULL, 0, "%d", i) : i);
      |                                      ~^   ~
      |                                       |   |
      |                                       int long int
      |                                      %ld
```

Reply:
Or would you like to just allow anything that silently promotes to long (so no casts at all).

Reply:
The diff from current would then be just the docs:
```
--- a/doc/man3/SSL_CTX_set1_curves.pod
+++ b/doc/man3/SSL_CTX_set1_curves.pod
@@ -33,7 +33,7 @@ SSL_get1_curves, SSL_get_shared_curve, SSL_CTX_get0_implemented_groups
  int SSL_get1_curves(SSL *ssl, int *curves);
  int SSL_get_shared_curve(SSL *s, int n);
 
- int SSL_CTX_get0_implemented_groups(SSL_CTX *ctx, long all,
+ int SSL_CTX_get0_implemented_groups(SSL_CTX *ctx, int all,
                                      STACK_OF(OPENSSL_CSTRING) *names);
 
 =head1 DESCRIPTION
```

Reply:
Done.